manageiq-operator/Dockerfile

```diff
@@ -1,5 +1,18 @@
 # Build the manager binary
-FROM docker.io/library/golang:1.23 as builder
+FROM registry.access.redhat.com/ubi9/ubi-minimal as builder
+
+COPY go.mod /go.mod
+
+RUN microdnf -y install gzip jq tar && \
+    arch=$(uname -m) && \
+    if [ ${arch} == "x86_64" ] ; then \
+      arch="amd64" \
+    ; fi && \
+    goversion=$(sed -nr 's/^go\s+(.*)/go\1/p' /go.mod) && \
+    file=$(curl -s 'https://go.dev/dl/?mode=json&include=all' | jq -r --arg GOVERSION "${goversion}" --arg ARCH "${arch}" 'map(select(.version | startswith($GOVERSION)))[0].files[] | select(.arch == $ARCH and .os == "linux").filename') && \
```

Comment:
Curious why you used map[0] over first? (Same in the end, but I figured first was clearer)

Reply:
`first` doesn't appear to do anything here...
`curl -s 'https://go.dev/dl/?mode=json&include=all' | jq -r 'map(select(.version | startswith("go1.23")))'`
and
`curl -s 'https://go.dev/dl/?mode=json&include=all' | jq -r 'first(map(select(.version | startswith("go1.23"))))'`
both return an array.

```
$ curl -s 'https://go.dev/dl/?mode=json&include=all' | jq -r 'first(map(select(.version | startswith("go1.23"))))'
[
  {
    "version": "go1.23.2",
    "stable": true,
    "files": [
      {
        "filename": "go1.23.2.src.tar.gz",
        "os": "",
        "arch": "",
        "version": "go1.23.2",
        "sha256": "36930162a93df417d90bd22c6e14daff4705baac2b02418edda671cdfa9cd07f",
        "size": 28171276,
        "kind": "source"
      },
.....
```

Reply:
Oh it's not first/map, it's

```
first(.[] | select(.version | startswith("go1.23.")))
```

Reply:
Done